pandas/tests/frame/test_constructors.py

```diff
@@ -1845,6 +1845,14 @@ def test_constructor_series_copy(self, float_frame):
 
         assert not (series["A"] == 5).all()
 
+    def test_object_array_does_not_copy(self):
+        a = np.array(["a", "b"], dtype="object")
```

Comment:
so if we have any columns that are inferred as datetimelike, we get copies of everything, i.e. dont retain any of the original array?

Reply:
If one of the columns is inferred as datetimelike, each column is created as its own block. From my understanding, upon consolidation, these blocks get copied to a contiguous location.

Reply:
add the issue number as a comment